lua/nvim-treesitter/configs.lua

```diff
@@ -442,6 +442,15 @@ function M.detach_module(mod_name, bufnr)
   end
 end
 
+-- Same as attach_module, but if the module is already attached, detach it first.
+-- @param mod_name the module name
+-- @param bufnr the bufnr
+-- @param lang the language of the buffer
+function M.reattach_module(mod_name, bufnr, lang)
+  M.detach_module(mod_name, bufnr)
```

Comment:
Can some detach function fail when the module wasn't attached before? Should we add a check whether the module is attached for `detach`?

Reply:
The detach function already checks for that https://github.com/nvim-treesitter/nvim-treesitter/blob/3413581b584f8fd124e299b57d77e1ee01886a05/lua/nvim-treesitter/configs.lua#L439-L439

Reply:
Someone already planned for that